pkg/templates/k8sresource_funcs_test.go

```diff
@@ -118,3 +119,186 @@ func TestFromConfigMap(t *testing.T) {
 		}
 	}
 }
+
+func TestCopySecretData(t *testing.T) {
+	t.Parallel()
+
+	testcases := []struct {
+		inputNs         string
+		inputSecretName string
+		inputKey        string
+		lookupNamespace string
+		expectedResult  string
+		expectedErr     error
+	}{
+		{"testns", "testsecret", "secretkey1", "", "secretkey1Val", nil},       // green-path test
+		{"testns", "testsecret", "secretkey2", "", "secretkey2Val", nil},       // green-path test
+		{"testns", "testsecret", "secretkey2", "testns", "secretkey2Val", nil}, // green-path test
+		{
+			"testns",
+			"idontexist",
+			"secretkey1",
+			"",
+			"secretkey2Val",
+			errors.New(`failed to get the secret idontexist from testns: secrets "idontexist" not found`),
+		}, // error : nonexistant secret
+		{"testns", "testsecret", "blah", "", "", nil}, // error : nonexistant key
+		{
+			"testns",
+			"testsecret",
+			"secretkey2",
+			"policies-ns",
+			"",
+			errors.New("the namespace argument passed to copySecretData is restricted to policies-ns"),
+		}, // error : restricted input namespace
+	}
+
+	for _, test := range testcases {
+		resolver, err := NewResolver(&k8sClient, k8sConfig, Config{LookupNamespace: test.lookupNamespace})
+		if err != nil {
+			t.Fatalf(err.Error())
+		}
+
+		val, err := resolver.copySecretData(test.inputNs, test.inputSecretName)
+
+		if err != nil {
+			if test.expectedErr == nil {
+				t.Fatalf(err.Error())
+			}
+
+			if !strings.EqualFold(test.expectedErr.Error(), err.Error()) {
+				t.Fatalf("expected err: %s got err: %s", test.expectedErr, err)
+			}
+		} else if !strings.Contains(val, base64encode(test.expectedResult)) {
```

Comment:
Optional: Ideally, you'd parse the JSON and ensure the key and value are correctly set, but this is good enough.

Reply:
fixed